File: backend/app/schemas/shelter.py
"""Nơi trú ẩn an toàn (điểm sơ tán) theo xã — có địa chỉ + toạ độ để chỉ đường."""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, Field


class ShelterKind(str, Enum):
    school = "school"                  # trường học
    community_hall = "community_hall"  # nhà văn hoá/sinh hoạt cộng đồng
    commune_office = "commune_office"  # trụ sở UBND xã
    health_station = "health_station"  # trạm y tế
    high_ground = "high_ground"        # điểm cao an toàn


class ShelterBase(BaseModel):
    commune_code: str
    name: str
    address: str
    lat: float
    lon: float
    capacity: int = Field(0, description="Sức chứa (người)")
    kind: ShelterKind = ShelterKind.community_hall
    contact_phone: str | None = None


class ShelterCreate(ShelterBase):
    pass


class Shelter(ShelterBase):
    id: str
    distance_km: float | None = Field(None, description="Khoảng cách tới điểm truy vấn (nếu có)")

File: frontend/utils/shelter.ts
import type { Shelter, ShelterKind } from "@/types";

export const SHELTER_KIND_LABELS: Record<ShelterKind, string> = {
  school: "Trường học",
  community_hall: "Nhà/trung tâm văn hóa",
  commune_office: "Trụ sở công cộng",
  health_station: "Cơ sở y tế",
  high_ground: "Khu sơ tán an toàn",
};

const MOCK_ESTIMATED_CAPACITY = 100;
const MOCK_ESTIMATED_CAPACITY_LABEL = "Khoảng 100–200 người";

export function mockShelterCapacity() {
  return MOCK_ESTIMATED_CAPACITY;
}

export function formatShelterCapacity(shelter: Pick<Shelter, "capacity" | "capacityStatus">) {
  if (shelter.capacityStatus !== "official") return MOCK_ESTIMATED_CAPACITY_LABEL;
  return `${shelter.capacity.toLocaleString("vi-VN")} người`;
}
